src/LuaTransformer.ts

```diff
@@ -141,11 +141,31 @@ export class LuaTransformer {
                     )
                 );
 
-                // return exports
-                statements.push(tstl.createReturnStatement([this.createExportsIdentifier()]));
+                if (this.options.outFile) {
+                    // package.loaded["sourceFile"] = exports
+                    statements.push(
+                        tstl.createAssignmentStatement(
+                            tstl.createTableIndexExpression(
+                                tstl.createTableIndexExpression(
+                                    tstl.createIdentifier("package"),
+                                    tstl.createStringLiteral("loaded")
+                                ),
+                                tstl.createStringLiteral(this.getExportPath(this.currentSourceFile))
+                            ),
+                            this.createExportsIdentifier()
+                        )
+                    );
+                } else {
+                    // return exports
+                    statements.push(tstl.createReturnStatement([this.createExportsIdentifier()]));
+                }
             }
         }
 
+        if (this.options.outFile) {
+            return [tstl.createBlock([tstl.createDoStatement(statements)], node), this.luaLibFeatureSet];
```

Comment:
We shouldn't wrap these in blocks when not in a module, as variables declared with let/const in one script should be able to be referenced from another.